Chemical reactions with FEniCS (#278)
precice/tutorials: channel-transport-reaction/README.md

---
title: Channel transport reaction
permalink: tutorials-channel-transport-reaction.html
keywords: FEniCS, Volumetric coupling, reaction, transport

> summary: We simulate chemical reactions in a moving fluid. One solver simulates the fluid flow and another handles the reactions.

Reviewer: ```suggestion
summary: We simulate an advection-diffusion-reaction scenario in a fluid in motion. One solver simulates the fluid flow and another handles the reactions.
```

---

{% note %}
Get the [case files of this tutorial](https://github.com/precice/tutorials/tree/master/channel-transport-reaction). Read how in the [tutorials introduction](https://www.precice.org/tutorials.html).
{% endnote %}

## Setup

> We solve a system of advection-diffusion-reaction in a fluid flowing past a cylinder. This case is inspired from [1]. A fluid flow inside a channel containing a cylinder. Chemical species A, B and C live in the fluid, with sources of A and B at the entry of the channel. Due to diffusion and fluid mixing, A and B react to form some C.

Reviewer: ```suggestion
We solve a system of advection-diffusion-reaction in a fluid flowing past a cylinder. This case is inspired from a FEniCS tutorial [1]. A fluid flows inside a channel containing a cylinder as an obstacle. Chemical species A, B and C live in the fluid, with sources of A and B at the entry of the channel. Due to diffusion and fluid mixing, A and B react to form a certain amount of C.
```

**Insert pic**

> The simulation is split into two participants: a Fluid participant that computes the fluid flow and sends the velocity field to the Chemical participant, computing the reactions, the diffusion and advection of all chemical species. The coupling is unidirectional (`serial-explicit` with only one data transfer).

Reviewer: ```suggestion
The simulation is split into two participants: a Fluid participant that computes the fluid flow and sends the velocity field to the Chemical participant which computes the diffusion and advection of all chemical species. The coupling is unidirectional (`serial-explicit` with only one data entity being transferred).
```

## Available solvers and dependencies

> Both participants run on FEniCS. Install [FEniCS](https://fenicsproject.org/download/) and the [FEniCS-adapter](https://github.com/precice/fenics-adapter).

Reviewer: ```suggestion
Both participants use the FEniCS library. Install [FEniCS](https://fenicsproject.org/download/) and [fenicsprecice](https://github.com/precice/fenics-adapter).
```

Author: Not a big fan of naming the link this way. Using "The X adapter" seems common in the rest of the documentation. See e.g. http://precice.org/tutorials-perpendicular-flap.html

Reviewer: Okay, lets keep this consistent and not make this change. Reading fenicsprecice is indeed a bit weird.
